Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -529,15 +530,18 @@ def splunk_submit_event_hec(hec_token, baseurl, event, fields, host, index, sour
     if hec_token is None:
         raise Exception('The HEC Token was not provided')
 
-    args = assign_params(
-        event=event,
-        host=host,
-        fields={'fields': fields} if fields else None,
-        index=index,
-        sourcetype=source_type,
-        source=source,
-        time=time_
-    )
+    args = {}
+    args["event"] = event
+    args["host"] = host
+    if fields:
+        args["fields"] = json.loads(fields)
+    args["index"] = index
+    args["sourcetype"] = source_type
+    args["source"] = source
+    if time_:
+        args["time"] = time_
+    else:
+        args["time"] = round(time.time())
```

Comment:
please explain the goal of this change

Reply:
Again, thank you for this too. We can use assign_params, but the way the fields param is parsed, is not correct according to what Splunk needs. Fields is supposed to contain key value pairs that is not part of the event.
Splunk reference: https://docs.splunk.com/Documentation/Splunk/8.0.3/Data/IFXandHEC
Snippet from that page:

> Assign the fields property to a JSON object that contains the custom fields to be indexed, as key-value pairs. For example, the following fields property, from within an HTTP request sent to the Splunk server, specifies two custom fields—club and wins:

So we should have something like this:
`args = assign_params(
        event=event,
        host=host,
        fields=json.loads(fields) if fields else None,
        index=index,
        sourcetype=source_type,
        source=source,
        time=time_
    )`